src/util/getElementType.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * @flow
+ */
+
+import type { JSXOpeningElement } from 'ast-types-flow';
+import { elementType } from 'jsx-ast-utils';
+
+import type { ESLintContext } from '../../flow/eslint';
+
+const getElementType = (context: ESLintContext): ((node: JSXOpeningElement) => string) => {
+  const { settings } = context;
+  const componentMap = settings['jsx-a11y']?.components;
+  if (!componentMap) {
+    return elementType;
+  }
+  return (node: JSXOpeningElement): string => {
+    const rawType = elementType(node);
+    return componentMap[rawType] || rawType;
```

Comment:
what about a custom component of `toString` etc? i guess we should handle that.
```suggestion
    return has(componentMap, rawType) ? componentMap[rawType] : rawType;
```
(with the `has` package)

Reply:
> what about a custom component of toString etc? 

I'm not sure I understand what is a custom component of `toString`

Note that the settings are coming from a JSON and thus are supposed to be plain. Can you give me a test case that would be handled differently by your suggestion please?

One I can think of is setting an empty string, although I'd say it's an abuse of the settings:
```json
"components": {
  "input": ""
}
```
That would silence all the rules alerting on `input` components.

Reply:
What i mean is, there could be jsx like this:
```jsx
<toString />
```
and `Object.prototype.toString` exists.

The settings can be plain, but the actual jsx won't be, nor is this JS code that reads the setting.

Reply:
OK, got it. I wasn't able to create a test that would actually be harmful, but that might be the case on older versions of Node. Done.